Sources/App/Controllers/PackageController+routes.swift

```diff
@@ -83,29 +83,27 @@ enum PackageController {
         }
     }
 
-    struct DocumentationVersion: Equatable {
-        var reference: Reference
-        var ownerName: String
-        var packageName: String
-        var docArchives: [String]
-        var latest: Version.Kind?
-        var updatedAt: Date
-    }
-
     static func defaultDocumentation(req: Request) async throws -> Response {
         guard
             let owner = req.parameters.get("owner"),
             let repository = req.parameters.get("repository")
         else {
             throw Abort(.notFound)
         }
+        let fragment = req.url.fragment.map { "#\($0)"} ?? ""
+        let path = req.parameters.getCatchall().joined(separator: "/") + fragment
 
-        let result = try await PackageResult.query(on: req.db, owner: owner, repository: repository)
-        if let documentationUrl = result.defaultDocumentationUrl {
-            throw Abort.redirect(to: documentationUrl)
-        } else {
+        guard let target = try await DocumentationTarget.query(on: req.db,
+                                                               owner: owner,
+                                                               repository: repository)
```

Comment:
This is probably the core change - push all the doc target queries into new type's `query` method so we can keep it all together a bit better and add some more focused tests.